### ISSUE TITLE
Creation Experience

### ISSUE THREAD

**Issue by harrycopeman** [CycleStudios/Robin#5]
What happens when a user clicks 'new'. We need to think of invoice and client creation all within the same or a similar space.

**Comment by wlenzenjr** [comment]
I see this going one of two directions. Either click the new, then click from the small dropdown what you want to create; or click new, then a popup appears where the entire experience is contained in what just popped up. 
![img_1276](https://cloud.githubusercontent.com/assets/9578832/13029763/baaf4de8-d25a-11e5-9a3d-f0d64329875b.JPG)
I personally like the second option because it's one less stop along the way to the creation box.

**New / Client**
- company name
- main contact name
- main contact contact info
- notes
- status (prospect, live, past)
- if status set to live:
  -- project name
  -- milestones (with +add milestones)
  -- completion date
  -- project cost
  -- invoice schedule

**New / Project**
- client
- project name
- milestones (with +add milestones)
- completion date
- project cost
- invoice schedule

**New / Invoice**
- client
- project
- date 
- invoice for (standard invoice info)
- amount
- set to recurring?
- send/export

**Comment by harrycopeman** [comment]
I also prefer the second option here. I'm wondering if we should not have the project and client thing merge into one. At the end of the day, if we had a client if was alway for one project, or if people on the occasion do have one client with 2 projects that will either a) manage it themselves in their task manager or b) create two 'clients' within Robin. It makes everything was easier for us and them. Here's my proposal:

#### New Client

company name
contact name
contact name info
notes/descriptipn
status (prospect, live, past) - user will create these and we will have presets
if status set to live: milestones (with +add milestones) -- completion date -- project cost -- invoice schedule <--- some of these optional

#### New Invoice

client
invoice number
date
invoice for (standard invoice info)
amount
set to recurring?
send/export

**Comment by wlenzenjr** [comment]
Think about this:

I'm an existing user. I have a client I've worked with on a project already. They've reached back out and want to do a new project. I need to create a new project for this client. 
1. Go to client in Robin
2. Click new project
3. Enter project details

Or
1. Click new project
2. Enter project details (which includes client)

Which one is simpler?

**Comment by harrycopeman** [comment]
You're right, I was just struggling to imagine how this would work on the main dash and calendar. But I've got it, we treat 'clients' as basically contacts, with only contact info's associated to them, then we have projects associated to those clients. Projects will be shown in the main dash and calendar, and you just link projects to a client. Makes complete sense, sweeeeet. Here's the final:

#### New Client

name
contact name
contact contact info
notes (optional)

#### New Project

client
project name
status (prospect, live, past)
if status set to live: -- project name -- milestones (with +add milestones) -- completion date -- project cost -- invoice schedule

#### New / Invoice

client / project
invoice number
date
invoice for (standard invoice info)
amount
set to recurring?
send/export

**Comment by wlenzenjr** [comment]
Perfect :+1:

**Comment by harrycopeman** [comment]
Design it :wink: :art:

**Comment by wlenzenjr** [comment]
Haha! Will do :)

**Comment by wlenzenjr** [comment]
I've started this in Figma. No idea if it's a good start. :/
